src/atoms/SegmentControl/SegmentControlOption.native.js

```diff
@@ -0,0 +1,248 @@
+import { getIconNames } from '../../_helpers/icon';
+import Size from '../Size';
+import automation from '../../_helpers/automation-attributes';
+import { useSegmentControlContext } from './SegmentControlContext';
+
+const styles = {
+  option: {
+    backgroundColor({ theme, variant, selected }) {
+      switch (variant) {
+        case 'filled':
+          if (selected) {
+            return getColor(theme, 'primary.920');
+          }
+          return getColor(theme, 'tone.930');
+        case 'outlined':
+          if (selected) {
+            return getColor(theme, 'primary.930');
+          }
+          return 'transparent';
+        default:
+          return 'transparent';
+      }
+    },
+    underlayColor({ theme, selected }) {
+      if (selected) {
+        return getColor(theme, 'primary.950');
+      }
+      return getColor(theme, 'tone.950');
+    },
+    borderBottomWidth({ variant }) {
+      if (variant === 'filled') {
+        return makePxValue(0.125);
+      }
+      return '0px';
+    },
+    borderColor({ theme, variant, selected, disabled }) {
+      if (variant === 'filled') {
+        if (selected && disabled) {
+          return getColor(theme, 'primary.400');
+        } else if (disabled) {
+          return getColor(theme, 'shade.920');
+        } else if (selected) {
+          return getColor(theme, 'primary.800');
+        } else {
+          return getColor(theme, 'shade.930');
+        }
+      }
+      return 'transparent';
+    },
+  },
+  text: {
+    color({ selected, disabled }) {
+      if (selected && disabled) {
+        return 'primary.500';
+      } else if (disabled) {
+        return 'shade.940';
+      } else if (selected) {
+        return 'primary.800';
+      } else {
+        return 'shade.960';
+      }
+    },
+    size({ size }) {
+      if (size === 'small') {
+        return 'small';
+      } else {
+        return 'medium';
+      }
+    },
+    weight({ variant }) {
+      if (variant === 'outlined') {
+        return 'bold';
+      } else {
+        return 'regular';
+      }
+    },
+  },
+  subText: {
+    color({ selected, disabled }) {
+      if (selected && disabled) {
+        return 'primary.400';
+      } else if (disabled) {
+        return 'shade.930';
+      } else if (selected) {
+        return 'primary.700';
+      } else {
+        return 'shade.950';
+      }
+    },
+  },
+  divider: {
+    backgroundColor({ theme, hide }) {
+      return hide ? 'transparent' : theme.colors.shade[930];
+    },
+  },
+  icon: {
+    size({ size }) {
+      if (size === 'small') {
+        return 'xsmall';
+      } else {
+        return 'small';
+      }
+    },
+  },
+};
+
+const StyledOption = styled(TouchableHighlight)`
+  background-color: ${styles.option.backgroundColor};
+  border-bottom-width: ${styles.option.borderBottomWidth};
+  border-color: ${styles.option.borderColor};
+`;
+
+const Divider = styled(View)`
+  position: absolute;
```

Comment:
Why 🙊

Reply:
Because I don't want the divider to take up actual space. It needs to appear and disappear without affecting the component